driver/lock.go

```diff
@@ -0,0 +1,194 @@
+package main
+
```

Comment:
Overall comment...

Have you considered making this part of Infoblox-go-client? IOW, do you think the NVLocker can be of more general use? For example, would this be useful for the CNI plugin too?